core/appmodule/genesis.go

```diff
@@ -0,0 +1,20 @@
+package appmodule
+
+import (
+	"encoding/json"
+	"io"
+
+	"google.golang.org/protobuf/runtime/protoiface"
+)
+
+type GenesisSource interface {
+	ReadMessage(protoiface.MessageV1) error
+	OpenReader(field string) (io.ReadCloser, error)
+	ReadRawJSON() (json.RawMessage, error)
```

Comment:
is `ReadMessage` and `ReadRawJSON` going to read next record or all records? If latter, then we should have kind of an iterator pattern:.